src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJson.php

```diff
@@ -65,12 +65,8 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
         $composerJsonContent = file_get_contents($composerJsonPath);
         assert(is_string($composerJsonContent));
 
-        /** @psalm-var array{autoload: ComposerAutoload}|null $composer */
-        $composer = json_decode($composerJsonContent, true);
-
-        if (! is_array($composer)) {
-            throw FailedToParseJson::inFile($composerJsonPath);
-        }
+        /** @psalm-var array{autoload: ComposerAutoload} $composer */
+        $composer = json_decode($composerJsonContent, true, flags: JSON_THROW_ON_ERROR);
```

Comment:
Mostly interested if `psalm --stats` says that this will improve our types or not: I believe `mixed` is still considered here.